andela/shadowcat-backend
# #167729785 Signup email verification

#### What does this PR do?
Users can receive verification email during signup

#### Description of Task to be completed?
1. Create user signup endpoint POST /api/v1/auth/signup
2. Create verification endpoint PUT /api/v1/auth/verify/:token

#### How should this be manually tested?
- Install dependencies with `npm install`
- Startup the localhost server with `npm run dev`
- On postman, Access the post route with the following body
  
![Screenshot (178)](https://user-images.githubusercontent.com/16121915/63558099-6df89880-c543-11e9-938c-b5986f596bdb.png)
- Hit send, and you should get a response similar to a sample below
![Screenshot (179)](https://user-images.githubusercontent.com/16121915/63558158-a8623580-c543-11e9-897e-de23d4d17fed.png)
- You should receive a verification email similar to this
![Screenshot (180)](https://user-images.githubusercontent.com/16121915/63558242-07c04580-c544-11e9-99ed-5302bf906dc8.png)

#### Any background context you want to provide?
- Moved documentation to public/api-docs as shown in the [simulations document](https://docs.google.com/document/d/1gaagRK8SpR7LM-5j9zuvUjq7mOFKle-0YyKsEjD5E84/edit) folder structure
- Added a middleware to check if the user had been validated
- Added `SENDGRID_API_KEY` and `WEB_MAIL_URL`, EMAIL_VERIFY_URL_TEST, EMAIL_VERIFY_URL_DEV, EMAIL_VERIFY_URL_PROD variables to your env file. Please contact me to get it.
- https://github.com/andela/shadowcat-backend/blob/b8800aab68082382953b9b9c650d54d75116b967/src/database/migrations/20190821140038-create-user.js#L27-L30
The allowNull property was set to true as requested by @sebaztyn as this allows him to store user details gotten from Google and Facebook. The details at his disposal do not have users' passwords. 

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/167729785

#### Screenshots (if appropriate)
None

#### Questions:
None
